### ISSUE TITLE
Segfault with Mesa 20.0.7

### ISSUE THREAD

**Issue by rharish101** [yshui/picom#411]
### Platform
Arch Linux

### GPU, drivers, and screen setup
* CPU: Intel 6700HQ
* iGPU: Intel HD Graphics 530
* dGPU: NVIDIA GTX 960M (could not test, as mesa broke these drivers too)

I was running picom in intel mode using [optimus-manager](https://github.com/Askannz/optimus-manager).

### Environment
WM: i3-gaps
DE: Xfce without xfdesktop4 and some other Xfce utilitiesqq

### picom version
vgit-e553e

### Configuration
* My main config: [picom.conf.txt](https://github.com/yshui/picom/files/4638731/picom.conf.txt)
* Arch Linux's default config: [default.conf.txt](https://github.com/yshui/picom/files/4638732/default.conf.txt)

### Steps of reproduction
1. Run picom as it is: `picom`
2. This should crash with a segfault.
3. Run picom without any config: `picom --config /dev/null`
4. This should work.

### Expected behavior
picom launches and works.

### Current Behavior
Segmentation fault occurs. I tried my custom config and also tried Arch Linux's default config. Behavior is similar to #324 .

### Stack trace
```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffee367700 (LWP 18138)]
[New Thread 0x7fffedb66700 (LWP 18139)]
[New Thread 0x7fffed365700 (LWP 18140)]
[New Thread 0x7fffecb64700 (LWP 18141)]

Thread 1 "picom" received signal SIGSEGV, Segmentation fault.
0x00007ffff5f3f995 in ?? () from /usr/lib/dri/iris_dri.so
#0  0x00007ffff5f3f995 in ?? () from /usr/lib/dri/iris_dri.so
#1  0x00007ffff5eca06e in ?? () from /usr/lib/dri/iris_dri.so
#2  0x00007ffff5ec9259 in ?? () from /usr/lib/dri/iris_dri.so
#3  0x00007ffff5ecf607 in ?? () from /usr/lib/dri/iris_dri.so
#4  0x00007ffff68ad402 in ?? () from /usr/lib/dri/iris_dri.so
#5  0x00007ffff5e836d3 in ?? () from /usr/lib/dri/iris_dri.so
#6  0x00007ffff5d87825 in ?? () from /usr/lib/dri/iris_dri.so
#7  0x00007ffff5d92275 in ?? () from /usr/lib/dri/iris_dri.so
#8  0x00007ffff5df25ab in ?? () from /usr/lib/dri/iris_dri.so
#9  0x000055555556dc4c in ?? ()
#10 0x0000555555576652 in ?? ()
#11 0x00005555555872b5 in ?? ()
#12 0x0000555555586fd4 in ?? ()
#13 0x0000555555589639 in ?? ()
#14 0x00007ffff7c7a033 in ev_invoke_pending () from /usr/lib/libev.so.4
#15 0x00007ffff7c7d902 in ev_run () from /usr/lib/libev.so.4
#16 0x000055555555e370 in ?? ()
#17 0x00007ffff7df3002 in __libc_start_main () from /usr/lib/libc.so.6
#18 0x000055555555f2ce in ?? ()
```

### Other logs
Output of `picom --log-level TRACE`: [logs.txt](https://github.com/yshui/picom/files/4638640/logs.txt)

**Comment by rharish101** [comment]
Sorry, this was fixed by a new release of mesa 20.0.7 by Arch Linux (20.0.7-2 over 20.0.7-1).

**Comment by yshui** [comment]
Np, glad it's fixed. :+1:
